package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -371,11 +373,21 @@ cdef class DCDFile:
             raise IOError('File opened in mode: {}. Writing only allowed '
                                'in mode "w"'.format('self.mode'))
 
+        #cdef double [:,:] unitcell = box
+        cdef DTYPE_t[::1] x = xyz[:, 0]
+        cdef DTYPE_t[::1] y = xyz[:, 1]
+        cdef DTYPE_t[::1] z = xyz[:, 2]
+
 	# prerequisite is a file struct for which the dcd header data
 	# has already been written
         self._write_header()
+
 	
-        #ok = write_dcdstep(self.fp, int curstep, int curframe,
-                         #self.n_atoms, <DTYPE_t*> &x[0],
-                         #<DTYPE_t*> &y[0], <DTYPE_t*> &z[0],
-                         #<DTYPE_t*> unitcell.data, self.charmm)
+        if self.current_frame == 0:
+            self.n_atoms = xyz.shape[0]
+            #self.box = box
+
+        ok = write_dcdstep(self.fp, step, self.current_frame,
+                         self.n_atoms, <DTYPE_t*> &x[0],
+                         <DTYPE_t*> &y[1], <DTYPE_t*> &z[2],
+                         <DTYPE_t2*> &box[0], charmm)
```

Comment:
Thanks. This line uncovered a pointer bug in the dcd reader library we use.  It has several issues. 

1. You say that the box pointer is 8 bit, but the underlying data is only 4bit. This means in total you have allocated 24 bit instead of the 48 that the dcd library except. Before you pass the pointer here you will need to convert the box type explicitly to a double. 

2. The readdcd.h file also doesn't pay attention to the correct pointer size. You will need to change the pointer from a float to a double throughout the readdcd.h file.

If we fix this there is one memory corruption bug less in MDAnalysis